### PR TITLE
Mn mk welcome prompt

### LINKED ISSUE #7: 7. As a user, I want to see a welcoming prompt to add my first item if my list is empty to help me get oriented to how the app works.
## Summary

When a user starts a new list, they need a little guidance on how to get started. Think through what you'd want to see as a user and make some intentional user experience decisions that will guide our users in confidently adding their very first item. 

## Acceptance criteria
- [ ] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

## Notes
- [The wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-empty.png) is a little short on the "welcoming" part of the story, but otherwise is functionally complete for the purposes of the AC.

### PR DESCRIPTION
**Pull request by michellerenehey**
## Description

This PR includes updates that help a user identify what needs to happen in the application. The code that we wrote includes a welcoming prompt to add a first item if a user's list is empty and a button to add an item. We conditionally render the search functionality for the list items, only showing the search (and clear search) capabilities if there are items in the list (in other words: a user does not see the search functionality if their list is empty). We also added a tiny amount of styling (centering the words on the page) for easy readability. 

## Related Issue
Closes #7 

## Acceptance Criteria
- [x] The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|     | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |
|   ✓  | :nail_care: Styling / UX    ✨ **NEW** ✨           |

### Before
list component without welcome message

### After
list component with welcome prompt

## Testing Steps / QA Criteria

1. Checkout the branch "MN-MK-welcome-prompt" and run `npm run`
2. It will be important to not have an assigned token to test this feature. So, if needed, go into your dev tools and clear your local storage token. 
3. From `/home`, click `Create New List` to generate a new token and be redirected to the `List` component.
4. You should see a brief `loading` state, followed by the welcome prompt telling you that your list is empty. _An important note for our fellow devs: we are open to feedback about changing this language and styling! This was/is just a first go at it._
5. From there, click `Add Item` to be redirected to `/additem`. Add an item and navigate back to your list by clicking the `List` NavLink (at the bottom of the screen). You will be redirected to `/list` and you should no longer see the welcome prompt.

### CODE REVIEW

**Review by N-Zubko — APPROVED** [review]
Hi @mira-kine and @michellerenehey, your PR is approved by me, everything is working exactly as described, great work!

As for the design - I personally do not like lots of bold letters, may be we could change styling in css it to `font-weight: normal `? 
In the loading state I would add a couple of words to make it sound warmer - "loading your shopping list" and we could an animated loader or spinner at a later stage.

I also noticed that the same bug I talked about last week, which became more confusing with the loading state - when there is no token in the local storage, the` /list` page keeps showing one word, which is "loading", however conditional button disabling could solve this issue.

File: src/App.jsx [hunk 1]
```diff
@@ -10,6 +10,8 @@ import { generateToken } from '@the-collab-lab/shopping-list-utils';
 export function App() {
 	const navigateTo = useNavigate();
 	const [data, setData] = useState([]);
+	const [loading, setLoading] = useState(true);
```
Comment (lindseyindev) [inline src/App.jsx:13]: niceeeee, adding a loader!

File: src/views/List.jsx [hunk 2]
```diff
@@ -27,38 +30,62 @@ export function List({ data }) {
...
+								</label>
+							</form>
+						</>
 					) : (
-						''
-					)}
-				</label>
-			</form>
+						<>
+							<h2>
```
Comment (lindseyindev) [inline src/views/List.jsx:72]: Nice work on adding this and the logic for the conditional render on this page!

**Review by lindseyindev — APPROVED** [review]
Great work on this issue! I liked that you added the loader mechanism and set it in the streamListItems function and pass it down! I was wondering if we should do anything to guard the data.length in the conditional render but your loader should serve as that guard -- was that your inspiration behind adding it? Would love to know your thought process!

**Review by Trevor-Rezac — APPROVED** [review]
I think everything looks great! My only suggestion somewhat echoes what Nadia suggested about making the font size smaller. Especially when viewing the app from any type of mobile view, the text just looks overly large.

**Comment by michellerenehey** [comment]
> Hi @mira-kine and @michellerenehey, your PR is approved by me, everything is working exactly as described, great work!
> 
> As for the design - I personally do not like lots of bold letters, may be we could change styling in css it to `font-weight: normal `? In the loading state I would add a couple of words to make it sound warmer - "loading your shopping list" and we could an animated loader or spinner at a later stage.
> 
> I also noticed that the same bug I talked about last week, which became more confusing with the loading state - when there is no token in the local storage, the` /list` page keeps showing one word, which is "loading", however conditional button disabling later on could solve this issue.

Thank you so much for the feedback, Nadia! I've changed the headers for both "Find what you're looking for!" and "Your list is empty! Click the button below to start building your list." to `<h3>` so they are a little bit smaller! I do think we will want to leave those as header tags, though, for accessibility. We can certainly make them less bold during our app styling further down the road. 

For the loading state, I've updated the language to be a little bit softer (thanks for that suggestion).

**Comment by michellerenehey** [comment]
> I think everything looks great! My only suggestion somewhat echoes what Nadia suggested about making the font size smaller. Especially when viewing the app from any type of mobile view, the text just looks overly large.

Thank you for the feedback! I addressed it in a comment to Nadia - certainly open to changing that styling in the future!

File: src/views/List.jsx (hunk 2 above)
Comment (michellerenehey) [reply src/views/List.jsx]: 💃  Thank you, Lindsey!

File: src/App.jsx (hunk 1 above)
Comment (michellerenehey) [reply src/App.jsx:13]: > Great work on this issue! I liked that you added the loader mechanism and set it in the streamListItems function and pass it down! I was wondering if we should do anything to guard the data.length in the conditional render but your loader should serve as that guard -- was that your inspiration behind adding it? Would love to know your thought process!

Yessss! Would love to speak to that in today's demo! We had finished up the acceptance criteria, but noticed that every time we hit the page we saw a quick 1-second glance of the search functionality (which we had hid behind some conditional rendering, as you noted previously)! Our thought was that the page was loading before it knew whether or not there was data in there (which is the base of the logic behind that render). How do you stop a component from rendering before it knows if it should? Loading state! When we added that, the problem was solved. 🙌

**Pull request merged**